cdap-common/src/main/java/co/cask/cdap/common/io/Locations.java

```diff
@@ -341,18 +342,22 @@ public static Location getParent(Location location) {
   }
 
   /**
-   * Returns the relative path for a given location URI
+   * Returns the relative path for a given locationURI.
+   * For failure to create URI it simply returns the Path from the given locationURI
    * @param locationFactory locationFactory to be used to create base URI
    * @param locationURI URI to be used for base URI
    * @return
    */
   public static String getRelativePath(LocationFactory locationFactory, URI locationURI) {
-    URI baseURI = locationFactory.create("").toURI();
-    URI relativeURI = baseURI.relativize(locationURI);
-    return relativeURI.getPath();
+    try {
+      URI baseURI = new URI(locationFactory.create("").toURI().getPath());
```

Comment:
since you know these are always valid because they come from URI objects, you can use URI.create() to avoid dealing with the checked URISyntaxException.